### ISSUE TITLE
showDatePicker text clipped [arabic-locale]

### ISSUE THREAD

**Issue by KirolousNashaat** [flutter/flutter#28778]
## Steps to Reproduce
1. set the locale of the app to arabic (right ot left)
2. use date picker dialog 
* the current selected date (ex: الجمعة, 1 يناير ) is clipped because of the month selection
some devices clip more than the others ... i think it have something to do with the arabic text size itself ?

```
flutter doctor -v
[√] Flutter (Channel stable, v1.2.1, on Microsoft Windows [Version 6.3.9600], locale en-US)
    • Flutter version 1.2.1 at E:\Setups\FLUTTER\flutter
    • Framework revision 8661d8aecd (2 weeks ago), 2019-02-14 19:19:53 -0800
    • Engine revision 3757390fa4
    • Dart version 2.1.2 (build 2.1.2-dev.0.0 0a7dcf17eb)

[√] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at E:\Setups\Android
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = E:\Setups\Android
    • Java binary at: E:\Sources\android-studio-ide-181.4987877-windows\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
    • All Android licenses accepted.

[!] Android Studio (version 3.2)
    • Android Studio at E:\Sources\android-studio-ide-181.4987877-windows\android-studio
    X Flutter plugin not installed; this adds Flutter specific functionality.
    • Dart plugin version 181.5656
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

[√] IntelliJ IDEA Community Edition (version 2018.2)
    • IntelliJ at E:\Setups\JetBrains\IntelliJ IDEA Community Edition 2018.2.3
    • Flutter plugin version 31.3.3
    • Dart plugin version 182.5124

[!] Connected device
    ! No devices available

! Doctor found issues in 2 categories.
```

**Comment by zoechi** [comment]
Can you please post a minimal reproductionß

**Comment by KirolousNashaat** [comment]
1. use the gallery sample as a project
2. add this lines :
pubspec :
''' flutter_localizations:
    sdk: flutter'''

app.dart :
'''
import 'package:flutter_localizations/flutter_localizations.dart';
 and this in material app :
localizationsDelegates: GlobalMaterialLocalizations.delegates,
        supportedLocales: [
          const Locale('ar'),
        ],
'''

3. build and navigate to date picker (in material) ,, look at the text at the beginning (after the year) , it will be clipped

**Comment by KirolousNashaat** [comment]
btw it 's different from device to another ,, sometimes it clip like 80% of the text , sometimes just 10% , varies alot
sry for my bad english

**Comment by darrenaustin** [comment]
This should be fixed with #31514.
